sdk/cpp/core/src/netconf_client_base.hpp

```diff
@@ -0,0 +1,49 @@
+/// YANG Development Kit
+// Copyright 2017 Cisco Systems. All rights reserved
+//
+////////////////////////////////////////////////////////////////
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+//////////////////////////////////////////////////////////////////
+
+#ifndef _YDK_NETCONF_CLIENT_BASE_H_
+#define _YDK_NETCONF_CLIENT_BASE_H_
+
+#include <string>
+#include <vector>
+
+namespace ydk
+{
+
+class NetconfClientBase
+{
+public:
+    NetconfClientBase() {}
```

Comment:
Can we avoid inlining constructors and destructors? [This](https://www.chromium.org/developers/coding-style/cpp-dos-and-donts#TOC-Stop-inlining-constructors-and-destructors) seems like a good guideline